script/Deploy.s.sol

```diff
@@ -3,14 +3,14 @@
 
 pragma solidity 0.8.20;
 
-import "forge-std/Script.sol";
-import {Counter} from "src/Counter.sol";
+import {Script} from "forge-std/Script.sol";
+import {ERC5564Announcer} from "src/ERC5564Announcer.sol";
 
 contract Deploy is Script {
```

Comment:
Is it worth writing a deploy script like this, and integrating it into the tests, when we know the real contract MUST be deployed according to Nick's method at a pre-computed address?

Reply:
Was chatting with @nerolation about this and current plan is to deploy via the default create2 deployer used by forge, which is [this one](https://github.com/Arachnid/deterministic-deployment-proxy). The rationale is:
- Nick's method requires not signing over chain ID, but some chains (e.g. Celo) require you to sign over chain ID, so it'd be impossible to deploy on that chain.
- Similarly, it requires hardcoding a gas price and gas limit but some chains (e.g. Arbitrum) have different gas limits/prices than other chains, so now you'd need a super high gas limit which isn't great
- Therefore deploying on those chains would require the chains to administratively place the ERC contract code there, which isn't likely.
- But, both of those chains already administratively placed Nick's deterministic deployer at the expected address. In other words, for chains where Nick's method doesn't work, the contract needs to be placed there, and you're more likely to get them to place (and some already have) a generic create2 deployer than a specific EIP's contracts.

Given that plan, I do think it makes to handle the deploy script this way.